experiments/conductor/cmd/runner/mock_commands.go

```diff
@@ -160,25 +178,181 @@ func readScriptYaml(opts *RunnerOptions, branch Branch) {
 		return
 	}
 
+	// Read and parse the script file
 	data, err := os.ReadFile(scriptFullPath)
 	if err != nil {
-		fmt.Println("Error reading file:", err)
+		log.Printf("Error reading file %s: %v", scriptFullPath, err)
 		return
 	}
-	dataStr := string(data)
-	if !strings.Contains(dataStr, " create") {
-		fmt.Println("WARNING: Script doesn't contain a CREATE command")
+
+	// Parse YAML content
+	var scriptContent []map[string]string
+	err = yaml.Unmarshal(data, &scriptContent)
+	if err != nil {
+		log.Printf("Error parsing YAML from %s: %v", scriptFullPath, err)
+		return
 	}
-	if !strings.Contains(dataStr, " update") {
-		fmt.Println("WARNING: Script doesn't contain an UPDATE command")
+
+	// Check commands and collect warnings
+	var warnings []string
+	hasCreate := false
+	hasDelete := false
+	hasList := false
+
+	for _, line := range scriptContent {
+		if cmd, ok := line["exec"]; ok && strings.Contains(cmd, "gcloud") {
+			if strings.Contains(cmd, " create ") {
+				hasCreate = true
+			}
+			if strings.Contains(cmd, " delete ") {
+				hasDelete = true
+			}
+			if strings.Contains(cmd, " list ") {
```

Comment:
Nit: It's possible that the generated list command is followed by no parameter, so can we check `" list"` (no space after `list`) instead?

Reply:
done.